add option to encapsulate (#199)
fastify/fastify-plugin: README.md

})
```

#### Encapsulate

> You can optionally keep the plugin encapsulated.

Reviewer: Maybe link to the fastify docs about encapsulation?

})
```

#### Encapsulate

> You can optionally keep the plugin encapsulated, which is the default behavior if you weren't using `fastify-plugin`.

Reviewer 2: ```suggestion

By default, `fastify-plugin` breaks the [encapsulation](https://github.com/fastify/fastify/blob/HEAD/docs/Reference/Encapsulation.md) but you can optionally keep the plugin encapsulated.
```

This allows you to set the plugin's name and validate its dependencies without making the plugin accessible.

> Read more about encapsulation [here](https://github.com/fastify/fastify/blob/main/docs/Reference/Encapsulation.md).

Reviewer 2: ```suggestion

```
